src/contracts/ODSaviour.sol

```diff
@@ -1,12 +1,168 @@
+import {IODSaviour} from '../interfaces/IODSaviour.sol';
+import {ODSafeManager, IODSafeManager} from '@opendollar/contracts/proxies/ODSafeManager.sol';
+import {Math} from '@opendollar/libraries/Math.sol';
+import {Assertions} from '@opendollar/libraries/Assertions.sol';
+import 'forge-std/console2.sol';
 
-contract ODSaviour is Authorizable {
-  constructor() {}
+/**
+ * @notice Steps to save a safe using ODSaviour:
+ *
+ * 1. Protocol DAO => connect [this] saviour `LiquidationEngine.connectSAFESaviour`
+ * 2. Treasury DAO => enable specific vaults `ODSaviour.setVaultStatus`
+ * 3. Treasury DAO => approve `ERC20.approveTransferFrom` to the saviour
+ * 4. Vault owner => protect thier safe with elected saviour `LiquidationEngine.protectSAFE` (only works if ARB DAO enable vaultId)
+ * 5. Safe in liquidation => auto call `LiquidationEngine.attemptSave` gets saviour from chosenSAFESaviour mapping
+ * 6. Saviour => increases collateral `ODSaviour.saveSAFE`
+ */
+contract ODSaviour is AccessControl, IODSaviour {
+  using Math for uint256;
+  using Assertions for address;
 
-  function enableVault(uint256 _tokenId) external {}
+  // solhint-disable-next-line modifier-name-mixedcase
+  bytes32 public constant SAVIOUR_TREASURY = keccak256(abi.encode('SAVIOUR_TREASURY'));
+  bytes32 public constant PROTOCOL = keccak256(abi.encode('PROTOCOL'));
+
+  uint256 public liquidatorReward;
+
+  address public saviourTreasury;
+  address public protocolGovernor;
+  address public liquidationEngine;
+
+  IVault721 public vault721;
+  IOracleRelayer public oracleRelayer;
+  IODSafeManager public safeManager;
+  ISAFEEngine public safeEngine;
+  ICollateralJoinFactory public collateralJoinFactory;
+
+  mapping(uint256 _vaultId => bool _enabled) private _enabledVaults;
+  mapping(bytes32 _cType => IERC20 _tokenAddress) private _saviourTokenAddresses;
+
+  /**
+   * @param _init The SaviourInit struct;
+   */
+  constructor(SaviourInit memory _init) {
+    saviourTreasury = _init.saviourTreasury.assertNonNull();
+    protocolGovernor = _init.protocolGovernor.assertNonNull();
+    vault721 = IVault721(_init.vault721.assertNonNull());
+    oracleRelayer = IOracleRelayer(_init.oracleRelayer.assertNonNull());
+    safeManager = IODSafeManager(address(vault721.safeManager()));
+    liquidationEngine = ODSafeManager(address(safeManager)).liquidationEngine(); // todo update @opendollar package to include `liquidationEngine` - PR #693
+    collateralJoinFactory = ICollateralJoinFactory(_init.collateralJoinFactory.assertNonNull());
+    safeEngine = ISAFEEngine(address(safeManager.safeEngine()));
+    liquidatorReward = _init.liquidatorReward;
+
+    if (_init.saviourTokens.length != _init.cTypes.length) revert LengthMismatch();
+
+    // solhint-disable-next-line  defi-wonderland/non-state-vars-leading-underscore
+    for (uint256 i; i < _init.cTypes.length; i++) {
+      _saviourTokenAddresses[_init.cTypes[i]] = IERC20(_init.saviourTokens[i].assertNonNull());
+    }
```

Comment:
- Use `_grantRole` instead of setup`
- Add `_grantRole(DEFAULT_ADMIN_ROLE, msg.sender);`

Reply:
we are now using authorizable.